functions/src/processors.ts

```diff
@@ -0,0 +1,91 @@
+'use strict';
+/*
+ * Copyright 2021 Algolia
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { firestore } from 'firebase-admin/lib/firestore';
+import DocumentReference = firestore.DocumentReference;
+import GeoPoint = firestore.GeoPoint;
+import Timestamp = firestore.Timestamp;
+
+const processObject = objectVal => {
+  const payload = {};
+  for (const [key, val] of Object.entries(objectVal)) {
+    const [field, value] = processValue(key, val);
+    payload[field] = value;
+  }
+  return payload;
+}
+
+const processValue = (field, value) => {
+  if (value instanceof DocumentReference) {
+    return [field, processDocumentReference(value)];
+  } else if (value instanceof GeoPoint) {
+    // Algolia has a prescribed field name for Geo data.
+    return ['_geoloc', processGeoPoint(value)];
+  } else if (value instanceof Timestamp) {
+    return [field, processTimestamp(value)];
+  } else if (value instanceof Array) {
+    return [field, processArray(value)];
+  } else if (value instanceof Object) {
+    return [field, processObject(value)];
+  }
+  return [field, value]
+}
```

Comment:
this is the code that's not tested yet :)

Reply:
Actually it is.  It's part of the main function that is testing each process.   My assertions to validate the expected and actual account for these new field types.

```
---------------|---------|----------|---------|---------|-------------------
File           | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
---------------|---------|----------|---------|---------|-------------------
All files      |   89.51 |    76.67 |      80 |   89.44 |
 config.ts     |     100 |      100 |     100 |     100 |
 extract.ts    |   89.29 |       50 |     100 |   88.89 | 39,57,70
 index.ts      |   89.74 |       75 |     100 |   89.74 | 48,62,73,93
 logs.ts       |   82.61 |      100 |      60 |   82.61 | 35,39,43,63
 processors.ts |   89.47 |    83.33 |   77.78 |   89.47 | 37,54,67,80
 util.ts       |     100 |      100 |     100 |     100 |
 version.ts    |     100 |      100 |     100 |     100 |
---------------|---------|----------|---------|---------|-------------------
Test Suites: 1 passed, 1 total
Tests:       5 passed, 5 total
Snapshots:   0 total
Time:        4.313 s
```

Reply:
I spoke with the Firebase team on the GeoPoint unit test.  They will get back to me.